setup.py

```diff
@@ -1,17 +1,55 @@
 from setuptools import setup
 
+
+with open("README.md") as readme_in:
+    long_description = readme_in.read()
+
+
 setup(
     name="cdisutils",
-    version="0.1.1",
-    description="Miscellaneous utilities useful for interaction with CDIS systems.",
-    license="Apache",
+    setup_requires=["setuptools_scm"],
+    use_scm_version={"local_scheme": "dirty-tag"},
+    author="Center for Translational Data Science",
+    author_email="support@nci-gdc.datacommons.io",
+    description="Miscellaneous utilities useful for interaction with CTDS systems.",
+    url="https://github.com/NCI-GDC/cdisutils",
+    license="Apache-2.0",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    classifiers=[
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.5",
+        "License :: OSI Approved :: Apache Software License",
+    ],
+    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4',
     packages=["cdisutils"],
+    # Note that some of these ranges have generous lower bounds since some
+    # consumers of cdisutils might still expect older versions.
     install_requires=[
-        'setuptools',
-        'attrs==19.1.0',
-        'xmltodict==0.9.2',
-        'pyOpenSSL==16.2.0',
-        'openpyxl==2.4.0',
-        'ndg-httpsclient==0.4.3',
+        "boto~=2.36",
+        "boto3~=1.9",
+        "future>=0.16,<1.0",
+        # Get indexclient from PyPI.
+        # TODO: See if a newer indexclient would still work.
+        "indexclient~=1.5",
```

Comment:
Are we sure we want to pull this from pypi right now? This seems to be the uc-cdis version. We could pin a nci-gdc version in the dev-requirements.in file. That way this repo's test suite can use the right one. The applications consuming this repo can pin to whatever indexd/gdcdatamodel they want to satisfy this requirement.

https://pypi.org/project/indexclient/#history

Reply:
I am thinking the idea is to pull these from nexus, which has NCI-GDC's indexclient and gdcdatamodels there right now

Reply:
I was thinking pip-compile might be less friendly if I put a Git URL here because Git URLs don't support version ranges. However, it looks like I can pin a different indexclient hash in a downstream project, and pip-compile will use that one and ignore whatever the cdisutils `setup.py` says. So, I will supply Git URLs here instead.

I am not sure we are 100% ready to pull gdcdatamodel from Nexus in all builds, and we should rename indexclient to avoid conflicting with the uc-cdis package when we push it to Nexus.

Reply:
Actually, given that this is a public repository that I think is used by both GDC and Gen3 code, it might be better to stick to the indexclient in PyPI, and allow clients to override if needed. I'll think a little more. But I did confirm that we can have, e.g., data-tools pin a specific NCI-GDC indexclient hash, and pip-compile will use that.

Reply:
> it might be better to stick to the indexclient in PyPI, and allow clients to override if needed.

This also means that this repo's test suite will run against the pypi versions. The latest 1.x version of indexclient on pypi is 1.5.8 and [this is what we'll be missing](https://github.com/uc-cdis/indexclient/compare/1.5.8...NCI-GDC:1.5.10).

It looks like we lose the ability to update `hashes` and `size` of an indexd doc (on the client side), the `bulk_get_latest` method, and some magic methods for comparing docs.

Reply:
...I see a lot of things in uc-cdis that depend on cdisutils, but I don't see it actually being used. I will put NCI-GDC URLs and leave the egg names the same.